www/app/page.tsx

```diff
@@ -135,6 +140,44 @@ export default function Home() {
     error: _,
   } = useSWR(conversationId, messagesFetcher, { revalidateOnFocus: false });
 
+  const handleReactionAdded = async (
+    messageId: string,
+    reaction: Exclude<Reaction, null>,
+  ) => {
+    if (!userId || !conversationId) return;
+
+    const api = new API({ url: URL!, userId });
+
+    try {
+      await api.addReaction(conversationId, messageId, reaction);
+
+      // Optimistically update the local data
+      mutateMessages((currentMessages) => {
+        if (!currentMessages) return currentMessages;
+        return currentMessages.map((msg) => {
+          console.log(`msgs: ${JSON.stringify(currentMessages)}`);
+          if (msg.id === messageId) {
+            return {
+              ...msg,
+              metadata: {
+                ...msg.metadata,
+                reaction,
+              },
+            };
+          }
+          console.log(`after update: ${JSON.stringify(currentMessages)}`);
+          return msg;
+        });
+      }, false); // Set to false to avoid revalidation immediately
+
+      // Trigger a revalidation to ensure data consistency
+      mutateMessages();
```

Comment:
Might be misunderstanding how `swr` works, but should we just set it to true if we want to revalidate immediately afterwards anyways?